Classes/common/touchdb/TDBlobStore.m

```diff
@@ -187,6 +190,8 @@ - (NSInteger)deleteBlobsExceptWithKeys:(NSSet*)keysToKeep withDatabase:(FMDataba
             [NSData dataWithBytes:oneRow.key.bytes length:sizeof(oneRow.key.bytes)];
         if ([keysToKeep containsObject:curKeyData]) {
             // Do not delete blob. It is an exception.
+            [filesToKeep addObject:oneRow.blobFilename];
+
             continue;
         }
 
```

Comment:
`numDeleted` is only used in `garbageCollectAttachments` to check for non-zero return value and for the log message.

So how about changing the signature of this method to return true/false and put the log message where the deletion actually happens, in `deleteFilesExceptWithFilenames`?

At the moment the logic seems a bit silly, where it's "delete them here and maybe try to delete them over there as well"